pkg/cli/build.go

```diff
@@ -309,6 +309,15 @@ func GenBuildOptions(c *cobra.Command, inputArgs []string, iopts BuildOptions) (
 		if err != nil {
 			return options, nil, nil, fmt.Errorf("unable to parse value provided %q as --cache-ttl: %w", iopts.CacheTTL, err)
 		}
+		// If user explicitly specified `--cache-ttl=0s`
+		// it would effectively mean that user is asking
+		// to use no cache at all. In such use cases
+		// buildah can skip looking for cache entierly
+		// by setting `--no-cache=true` internally.
+		if int64(cacheTTL) == 0 {
+			logrus.Debug("Setting --no-cache=true since --cache-ttl was set to 0s which effectively means user wants to ignore cache")
+			iopts.NoCache = true
```

Comment:
for completeness, you should throw an error if user specified

==nocache=false
if iopts.NoChache==false &&  c.Flag("no-cache").Changed  {
            return fmt.Errorf(...)
}

Reply:
Yes I agree